### PR TITLE
Remove clang tools binaries from regular installation

### PR DESCRIPTION
**Pull request by jmillan**
Move them to worker/scripts

Bonus point: Remove Gulp dependency too from regular installation.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -29,6 +29,8 @@
 
 # Python generated stuff in Windows.
 /worker/scripts/configure.pyc
+# worker scripts node-modules.
+/worker/scripts/node_modules
```
Comment (ibc) [inline .gitignore]:
```suggestion
/worker/scripts/node_modules/
```

File: npm-scripts.js [hunk 2]
```diff
@@ -72,21 +72,21 @@ switch (task)
 	case 'lint:node':
 	{
 		execute('cross-env MEDIASOUP_NODE_LANGUAGE=typescript eslint -c .eslintrc.js --max-warnings 0 --ext=ts src/');
-		execute('cross-env MEDIASOUP_NODE_LANGUAGE=javascript eslint -c .eslintrc.js --max-warnings 0 --ext=js --ignore-pattern \'!.eslintrc.js\' .eslintrc.js gulpfile.js npm-scripts.js test/');
+		execute('cross-env MEDIASOUP_NODE_LANGUAGE=javascript eslint -c .eslintrc.js --max-warnings 0 --ext=js --ignore-pattern \'!.eslintrc.js\' .eslintrc.js npm-scripts.js test/');
```
Comment (ibc) [inline npm-scripts.js]: Why is gulpfile.js removed from lint?

File: worker/Makefile [hunk 3]
```diff
@@ -77,8 +77,8 @@ bear:
 tidy:
 	$(SED) "s|PATH|$(PWD)|g" compile_commands_template.json > compile_commands.json
 	$(PYTHON) ./scripts/clang-tidy.py \
-		-clang-tidy-binary=../node_modules/.bin/clang-tidy \
-		-clang-apply-replacements-binary=../node_modules/.bin/clang-apply-replacements \
+		-clang-tidy-binary=./scripts/node_modules/.bin/clang-tidy \
+		-clang-apply-replacements-binary=./scripts/node_modules/.bin/clang-apply-replacements \
```
Comment (ibc) [inline worker/Makefile:81]: Are those still work if I do `cd worker/ && make tidy`?

**Comment by ibc** [comment]
CI failing

**Comment by ibc** [comment]
So, in order for this to work locally, we must enter `worker/` and run `npm install` before running usual npm/make commands?

File: worker/Makefile (hunk 3 above)
Comment (jmillan) [reply worker/Makefile:81]: yes, it does.

**Comment by jmillan** [comment]
> So, in order for this to work locally, we must enter worker/ and run npm install before running usual npm/make commands?

Correct. Not enter `worker` but `worker/scripts`

**Comment by jmillan** [comment]
> CI failing

On it

**Comment by jmillan** [comment]
ooh, my...

File: npm-scripts.js (hunk 2 above)
Comment (jmillan) [reply npm-scripts.js]: It was previously *excluded*. Now it's moved to `worker/scripts`

File: .github/workflows/mediasoup-worker.yaml [hunk 4]
```diff
@@ -69,8 +69,8 @@ jobs:
 
       - run: npm install
       - run: |
-          pushd worker/scripts
+          cd worker/scripts
           npm install
-          popd
+          cd ../..
```
Comment (nazar-pc) [inline .github/workflows/mediasoup-worker.yaml]: I think it runs every command under the same directory regardless of `cd` in previous command...

Comment (jmillan) [reply .github/workflows/mediasoup-worker.yaml]: oh, makes sense. Thanks

Comment (jmillan) [reply .github/workflows/mediasoup-worker.yaml]: So probably this is executing `npm install` in the root directory again instead of in `worker/scripts`, right?

Comment (jmillan) [reply .github/workflows/mediasoup-worker.yaml]: Fixed, let's see

File: npm-scripts.js [hunk 5]
```diff
@@ -163,6 +161,13 @@ switch (task)
 		break;
 	}
 
+	case 'install-clang-tools':
```
Comment (ibc) [inline npm-scripts.js:164]: Why is this needed at all? Nobody calls `node npm-scripts.js install-clang-tools`.

Comment (ibc) [reply npm-scripts.js:164]: ok, yes, package.json does. What I mean is: does it make sense to call something from.... forget it, it's ok.

**Review by ibc — APPROVED** [review]
